app/src/main/java/com/nextcloud/talk/activities/FullScreenImageActivity.kt

```diff
@@ -53,7 +61,8 @@ class FullScreenImageActivity : AppCompatActivity() {
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return if (item.itemId == android.R.id.home) {
-            onBackPressed()
+            // onBackPressed()
+            onBackPressedDispatcher.addCallback(this, callback)
```

Comment:
Currently this back button behaviour won't work. 


the 
`onBackPressedDispatcher.addCallback(this, callback)`
should be set in `onCreate`.

Here you just have to call onBackPressedDispatcher.onBackPressed():

```
  override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (item.itemId == android.R.id.home) {
               onBackPressedDispatcher.onBackPressed()
```

Same for other activities where the home button is used in onOptionsItemSelected.